### PR TITLE
Ncf perf optimizations for CTL and multi GPU

### PR DESCRIPTION
**Pull request by nnigania**
Ncf perf changes 1)exclude metric layer from CTL train step 2) Dataset optimization to fix size of the sample_weights, preventing a costly broadcast during loss calculation for multi-gpu case

### CODE REVIEW

File: official/recommendation/ncf_keras_main.py [hunk 1]
```diff
@@ -224,7 +224,8 @@ def _get_keras_model(params):
       [zeros, logits],
       axis=-1)
 
-  softmax_logits = MetricLayer(params)([softmax_logits, dup_mask_input])
+  if not params["keras_use_ctl"]:
```
Comment (guptapriya) [inline official/recommendation/ncf_keras_main.py]: Add a comment as to why this is not neeeded in ctl case
